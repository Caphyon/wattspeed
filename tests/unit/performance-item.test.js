import { shallowMount } from "@vue/test-utils";
import speed_item from "../../src/components/panels/tech/items/performance-item"; // name of your Vue component

describe("performance-item.vue", () => {
  it("Global", () => {
    const wrapper = shallowMount(speed_item, {
      propsData: {
        data: {
          title: "Use video formats for animated content",
          msg: "Large GIFs are inefficient for delivering animated content",
          level: null
        }
      }
    });

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('div.scrollable--issue').exists()).toBe(true);
    expect(wrapper.find('h4').exists()).toBe(true);
    expect(wrapper.find('p').exists()).toBe(true);
    wrapper.destroy();
  });

  it("Snapshot for speed-item component level danger", () => {
    const wrapper = shallowMount(speed_item, {
      propsData: {
        data: {
          title: "Use video formats for animated content",
          msg: "Large GIFs are inefficient for delivering animated content",
          level: null
        }
      }
    });

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('div.scrollable--issue.alert--danger').exists()).toBe(true);
    wrapper.destroy();
  });

  it("Snapshot for speed-item component level warning", () => {
    const wrapper = shallowMount(speed_item, {
      propsData: {
        data: {
          title: "Use video formats for animated content",
          msg: "Large GIFs are inefficient for delivering animated content",
          level: 0.67
        }
      }
    });

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('div.scrollable--issue.alert--warning').exists()).toBe(true);
    wrapper.destroy();
  });

  it("Snapshot for speed-item component level success", () => {
    const wrapper = shallowMount(speed_item, {
      propsData: {
        data: {
          title: "Use video formats for animated content",
          msg: "Large GIFs are inefficient for delivering animated content",
          level: 1.0
        }
      }
    });

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    expect(wrapper.find('div.scrollable--issue.alert--warning').exists()).toBe(false);
    expect(wrapper.find('div.scrollable--issue.alert--danger').exists()).toBe(false);
    wrapper.destroy();
  });

});
