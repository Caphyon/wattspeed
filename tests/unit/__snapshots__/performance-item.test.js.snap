// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`performance-item.vue Global 1`] = `
<div
  class="scrollable--issue alert--danger"
>
  <h4
    class="mt0 mb0"
  >
    Use video formats for animated content
  </h4>
   
  <p
    class="mb0"
  >
    Large GIFs are inefficient for delivering animated content
  </p>
</div>
`;

exports[`performance-item.vue Snapshot for speed-item component level danger 1`] = `
<div
  class="scrollable--issue alert--danger"
>
  <h4
    class="mt0 mb0"
  >
    Use video formats for animated content
  </h4>
   
  <p
    class="mb0"
  >
    Large GIFs are inefficient for delivering animated content
  </p>
</div>
`;

exports[`performance-item.vue Snapshot for speed-item component level success 1`] = `
<div
  class="scrollable--issue alert--success"
>
  <h4
    class="mt0 mb0"
  >
    Use video formats for animated content
  </h4>
   
  <p
    class="mb0"
  >
    Large GIFs are inefficient for delivering animated content
  </p>
</div>
`;

exports[`performance-item.vue Snapshot for speed-item component level warning 1`] = `
<div
  class="scrollable--issue alert--warning"
>
  <h4
    class="mt0 mb0"
  >
    Use video formats for animated content
  </h4>
   
  <p
    class="mb0"
  >
    Large GIFs are inefficient for delivering animated content
  </p>
</div>
`;
